src/bssl.rs

```diff
@@ -12,12 +12,62 @@
 // OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 // CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-use {c, error};
+pub use self::ffi_error::ErrorCode;
 
-pub fn map_result(bssl_result: c::int) -> Result<(), error::Unspecified> {
-    match bssl_result {
-        1 => Ok(()),
-        _ => Err(error::Unspecified),
+/// Many foreign functions return a `c::int` indicating whether an error occurred. This module
+/// wraps that error code in a `#[repr(transparent)]` newtype which can be converted into a
+/// `Result`.
+mod ffi_error {
+    use core::marker::PhantomData;
+    use {c, error};
+
+    /// The meaning of the error code returned by a foreign function.
+    pub trait Semantics {
```

Comment:
I don't see the advantage of factoring this out into a trait. One of the design principles of *ring* is to avoid unnecessary abstractions, and at least the way the code is currently, this abstraction isn't necessary.

Reply:
Just two separate structs here then?

I had imagined that this would make it easy to share code later as features were added, but that's kind of the definition of overabstraction :)

Reply:
Also, how do you feel about using `Posix` to mean returns `0` on failure. It was the best, shortest thing I could think of, but `bssl::PosixResult` reads a bit strange.